src/include/reconcile_inline.h

```diff
@@ -195,11 +195,11 @@ __wt_rec_need_split(WT_RECONCILE *r, size_t len)
      * worth splitting. We also want to temper this effect to avoid in-memory updates from
      * dominating the calculation and causing excessive splitting. Therefore, we'll limit the impact
      * to a tenth of the cache usage occupied by those updates. For small updates the overhead of
-     * the update structure could skew the calculation, so we subtract the overhead before
-     * considering the cache usage by the updates.
+     * the update structure could skew the calculation. Therefore, we only consider the size of the
+     * value in the saved updates' memory size
      */
     if (r->page->type == WT_PAGE_ROW_LEAF && page_items > WT_REC_SPLIT_MIN_ITEMS_USE_MEM)
-        len += (r->supd_memsize - ((size_t)r->supd_next * WT_UPDATE_SIZE)) / 10;
+        len += r->supd_memsize / 10;
```

Comment:
It's worth noting that this calculation was pretty hard won. You can see the history in WT-6532 and WT-6487.

I suspect that the change here will mean that splits happen less often - which is the direction WT-6532 was going in anyway, so hopefully it's the right choice.

We don't have a direct test that will show a failure if this calculation changes, it will just alter the size of a database being created on disk, or the internal tree structure.

As such, this change deserves a set of sys-perf performance testing and analysis before merge.

Reply:
Yes, I've run the sys-perf test. The result is in the ticket. The results are green and I'm not sure which test in the suite I should look at in detail though.